packages/server/src/automations/steps/updateRow.ts

```diff
@@ -82,39 +82,73 @@ export async function run({ inputs, appId, emitter }: AutomationStepInput) {
   }
   const tableId = inputs.row.tableId
 
-  // clear any undefined, null or empty string properties so that they aren't updated
-  for (let propKey of Object.keys(inputs.row)) {
-    const clearRelationships =
-      inputs.meta?.fields?.[propKey]?.clearRelationships
-    if (
-      (inputs.row[propKey] == null || inputs.row[propKey]?.length === 0) &&
-      !clearRelationships
-    ) {
-      delete inputs.row[propKey]
-    }
+  // Base update
+  let rowUpdate: Record<string, any>
+
+  // Legacy
+  // Find previously set values and add them to the update. Ensure empty relationships
+  // are added to the update if clearRelationships is true
+  const legacyUpdated = Object.keys(inputs.row || {}).reduce(
```

Comment:
Just a question on this - when the new frontend next makes a change to this automation, will inputs.row be removed from the saved definition? Just noticing that `legacyUpdated` overrides `parsedUpdate`, so if it didn't get cleared then the saved values in `inputs.row` will forever override the new meta info.

I assume this does happen, but thought it was faster to ask than try to find the relevant code in the frontend haha.

Reply:
Yep, the second the user alters the automation in the frontend, everything that shouldn't be present is purged. The legacy stuff in the server is just there to accomodate any unchanged automations. That's why it iterates over the `inputs.row` and not `inputs.meta.fields`.